### PR TITLE
LR fields constructor from spline objects

### PR DESCRIPTION
**Pull request by akva2**
Instead of only supporting objects hosted in ASM classes. Required for recovery purposes.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-openfrac=19 please

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-openfrac=19 please

File: src/ASM/LR/LRSplineFields3D.h [hunk 1]
```diff
@@ -42,6 +42,14 @@ class LRSplineFields3D : public Fields
   LRSplineFields3D(const ASMu3D* patch, const RealArray& v,
                    char basis = 1, int nf = 0, const char* name = nullptr);
 
+  //! \brief Construct directly from surface.
```
Comment (kmokstad) [inline src/ASM/LR/LRSplineFields3D.h]: You mean volume

File: src/ASM/LR/LRSplineFields3D.h [hunk 2]
```diff
@@ -42,6 +42,14 @@ class LRSplineFields3D : public Fields
...
+  //! \param[in] srf The spline surface to use
```
Comment (kmokstad) [inline src/ASM/LR/LRSplineFields3D.h]: spline volume, and rename the variable (I know, copy-paste..)

File: src/ASM/LR/LRSplineFields3D.C [hunk 3]
```diff
@@ -49,6 +49,15 @@ LRSplineFields3D::LRSplineFields3D (const ASMu3D* patch,
 }
 
 
+LRSplineFields3D::LRSplineFields3D (const LR::LRSplineVolume* srf,
```
Comment (kmokstad) [inline src/ASM/LR/LRSplineFields3D.C]: Rename the first argument, it is not a surface.

File: src/Utility/Test/TestFieldFunctionsLR.C [hunk 4]
```diff
@@ -53,7 +53,7 @@ TEST(TestFieldFunctions, 2D1PLR)
   scal = f2D_scalar(X);
   x = 2.0, y = 0.5;
   vec = f2D_vec(X);
-  EXPECT_NEAR(scal, x, 1e-14);
+  EXPECT_NEAR(scal, x, 1e-13);
```
Comment (kmokstad) [inline src/Utility/Test/TestFieldFunctionsLR.C]: If the changes in this commit is due to the fix in the previous commit, it should be squashed into that one

**Comment by akva2** [comment]
construct from asm is still necessary and in use. it could of course be refactored to only have the single constructors, but then the code from those constructors have to go somewhere else so i don't think there would be much benefit.

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-openfrac=19 please

**Pull request merged**